### PR TITLE
Add search to tasks index page

### PR DESCRIPTION
**Pull request by arku**
Implements the feature requested in https://github.com/taskcluster/taskcluster/issues/1148

### CODE REVIEW

File: ui/src/views/Tasks/TaskIndex/IndexedTask/index.jsx [hunk 1]
```diff
@@ -90,6 +91,21 @@ export default class IndexedTask extends Component {
...
 
+  handleTaskNamespaceSearchSubmit = taskNamespaceSearch => {
+    const searchText = taskNamespaceSearch.replace(/\s+/g, '');
+    const namespace = searchText
+      .split('.')
+      .slice(0, -1)
+      .join('.');
+    const taskFromNamespace = searchText.split('.').slice(-1)[0];
+
+    if (namespace && taskFromNamespace) {
```
Comment (arku) [inline ui/src/views/Tasks/TaskIndex/IndexedTask/index.jsx]: @helfi92 Is this fine?

Here are the possible ways someone might interact with the search bar and the obtained results.

**No task in the namespace**

If a user just searches for 'gecko', nothing would happen as there is no task identified from the namespace.

**Namespace and task present**

If a user searches for `ci.cache.level-3.docker-images.v2.hg-push.latest`, they will be taken to `/tasks/index/ci.cache.level-3.docker-images.v2.hg-push/latest` which will render the `IndexedTask` component.

But if search for `ci.cache.level-3.docker-images.v2.hg-push`, they will be taken to `/tasks/index/ci.cache.level-3.docker-images.v2/hg-push` which won't work and an error will be rendered.

Should we worry about this? If yes, how do we handle this scenario?

File: ui/src/views/Tasks/TaskIndex/ListNamespaces/index.jsx [hunk 2]
```diff
@@ -138,7 +154,14 @@ export default class ListNamespaces extends Component {
     return (
       <Dashboard
         title="Index Browser"
-        helpView={<HelpView description={description} />}>
+        helpView={<HelpView description={description} />}
+        search={
```
Comment (arku) [inline ui/src/views/Tasks/TaskIndex/ListNamespaces/index.jsx]: @helfi92  If a user searches for a task that doesn't exist, the `IndexedTask` component will be rendered with the error message. If they want to retry (search again), they have go back to the Index page again. So, I decided to add the search bar here as well.

File: ui/src/views/Tasks/TaskIndex/IndexedTask/index.jsx (hunk 1 above)
Comment (helfi92) [reply ui/src/views/Tasks/TaskIndex/IndexedTask/index.jsx]: No worries!

**Comment by helfi92** [comment]
@arku Let me know when this is ready for review :)

**Comment by arku** [comment]
@helfi92 Will do.

**Pull request closed without merging**